index.bs

```diff
@@ -1967,13 +1995,22 @@ To <dfn>obtain a randomized source response</dfn> given a [=randomized response
 
 <h3 algorithm id="computing-channel-capacity">Computing channel capacity</h3>
 
-To <dfn>compute the channel capacity of a source</dfn> given a [=randomized response output configuration=] |config| and a double |epsilon|:
-1. Let |pickRate| be the [=obtain a randomized source response pick rate|randomized response pick rate=] with |config| and |epsilon|.
-1. Let |states| be the [=obtain a set of possible trigger states|number of possible trigger states=] with |config|.
-1. If |states| is 1, return 0.
-1. If |states| is greater than the user agent's [=max trigger-state cardinality=], return an error.
-1. Let |p| be |pickRate| * (|states| - 1) / |states|.
-1. Return log2(|states|) - h(|p|) - |p| * log2(|states| - 1) where h is the binary entropy function [[BIN-ENT]].
+To <dfn>compute the channel capacity of a source</dfn> given a [=randomized response output configuration=] |config|, a double |epsilon|, a possibly null positive 32-bit integer |attributionScopeLimit|, and a possibly null positive 32-bit integer |maxNumEventStates|:
+1. If |attributionScopeLimit| is null, set |attributionScopeLimit| to 1.
+1. If |maxNumEventStates| is null, set |maxNumEventStates| to 1.
+1. Let |maxInformationGain| be 0.
+1. [=list/iterate|For each=] integer |numUsedScopes| of [=the exclusive range|the range=] 0 to |attributionScopeLimit|, exclusive:
+    1. [=list/iterate|For each=] integer |numEventStates| of [=the range=] 1 to |maxNumEventStates|, inclusive:
+        1. Let |informationGain| be [=compute information gain=] given |config|, |epsilon|, |numUsedScopes| and |numEventStates|
+        1. If |informationGain| is greater than |maxInformationGain|, set |maxInformationGain| to |informationGain|.
+1. Return |maxInformationGain|.
+
+To <dfn>compute information gain</dfn> given a [=randomized response output configuration=] |config|, a double |epsilon|, a positive 32-bit integer |numUsedScopes| and a positive 32-bit integer |numEventStates|:
+1. Let |numTriggerStates| be the [=obtain a set of possible trigger states|number of possible trigger states=] with |config|.
+1. Let |totalStates| be |numTriggerStates| + |numUsedScopes| * |numEventStates|.
```

Comment:
Is this making the assumption that `numTriggerStates` is associated with a single attribution scope, because we are assigning `numUsedScopes` of possible `numEventStates`? But one source can have from 0 to multiple scopes.

Reply:
- `numTriggerStates` is the number of report states could be generated from the source, which represents q1 in the info gain formula. 
- `numUsedScopes` is essentially the number of possible "fake" events a bad actor can apply, it is `k` in the formula. 
- `numEventStates` is the possible states for single view / event, which represents `q2` in the formula.